e2e/actions/regressions.go

```diff
@@ -213,3 +218,100 @@ func (c actionTests) issue4836(t *testing.T) {
 		),
 	)
 }
+
+// Check that image caching from an http source works correctly, using
+// the HTTP Last-Modified header to invalidate previously pulled images,
+// and that if we exec c1/v1.sif and then c1/v2.sif the latter is *not*
+// run from the cached image of the former.
+func (c actionTests) issue4823(t *testing.T) {
+	e2e.EnsureImage(t, c.env)
+
+	// Copy image to a local tempdir so we can modify times on it
+	issueDir, cleanup := e2e.MakeTempDir(t, c.env.TestDir, "issue-4823-", "")
+	defer cleanup(t)
+	issueImage := path.Join(issueDir, "test.sif")
+	if err := fs.CopyFile(c.env.ImagePath, issueImage, 0755); err != nil {
+		t.Fatalf("Could not copy test image file: %v", err)
+	}
+
+	// Start an http server that always serves the same file from
+	// whatever URL is requested
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, issueImage)
+	}))
+	defer srv.Close()
+
+	tests := []struct {
+		name         string
+		urlPath      string
+		touchFile    bool
+		expectCached bool
+	}{
+		// First time we use an image at a URL it is not cached
+		{
+			name:         "http_c2v1.sif_uncached",
+			urlPath:      "/c1/v1.sif",
+			touchFile:    false,
+			expectCached: false,
+		},
+		// Second time at same URL is cached
+		{
+			name:         "http_c2v1.sif_cached",
+			urlPath:      "/c1/v1.sif",
+			touchFile:    false,
+			expectCached: true,
+		},
+		// Date of file at the URL is modified - not cached
+		{
+			name:         "http_c2v1.sif_touched_uncached",
+			urlPath:      "/c1/v1.sif",
+			touchFile:    true,
+			expectCached: false,
+		},
+		// Different URL - not cached
+		{
+			name:         "http_c2v2.sif_uncached",
+			urlPath:      "/c1/v2.sif",
```

Comment:
Should this url be `/c2/v1.sif` instead? If the cache was keyed with just the last part of the url, as it was previously, then wouldn't cause a collision if I understand the issue correctly

Reply:
Yes - you are correct. Good spot, and :man_facepalming: from me seeing as this is what the test is meant to be for